client/my-sites/sidebar/sidebar.jsx

```diff
@@ -681,10 +678,16 @@ export const MySitesSidebar = React.createClass( {
 	},
 
 	render: function() {
-		var publish = !! this.publish(),
-			appearance = ( !! this.themes() || !! this.menus() ),
-			configuration = ( !! this.sharing() || !! this.users() || !! this.siteSettings() || !! this.plugins() || !! this.upgrades() ),
-			vip = !! this.vip();
+		const showPublishMenu = !! this.renderPublishMenu();
+		const showAppearanceMenu = !! this.renderThemesOption() || !! this.renderMenusOption();
+		const showConfigurationMenu = (
+			!! this.renderSharingOption() ||
+			!! this.renderUsersOption() ||
+			!! this.renderSettingsOption() ||
+			!! this.renderPluginsOption() ||
+			!! this.renderUpgradesOption()
+		);
```

Comment:
> I usually prefer to do the checking separately first and just not even attempt to call this function (or component once it's moved out) if those checks aren't true.

We do prefer early returns. In this case I think that the logic is complex enough to decompose pieces into additional components that may render nothing if we don't meet those checks.